### ISSUE TITLE
unserialize(): Error at offset 0 of 56 bytes

### ISSUE THREAD

**Issue by joostvanhoof** [spatie/mailcoach-support#235]
I get `unserialize(): Error at offset 0 of 56 bytes` multiple times when I'm creating and sending a campaign. Screenshot below of where it occurs, and these are the places that trigger the error:

```
\Spatie\Mailcoach\Http\App\Controllers\Campaigns\Draft\SendCampaignController
\Spatie\Mailcoach\Http\App\Controllers\Campaigns\Draft\CampaignDeliveryController
\Spatie\Mailcoach\Http\App\Controllers\Campaigns\Draft\CampaignSettingsController
```

<img width="1141" alt="Screenshot 2020-08-18 at 17 23 27" src="https://user-images.githubusercontent.com/5268113/90532761-e9e9e880-e177-11ea-86e9-605df32cc4cb.png">

**Comment by riasvdv** [comment]
Is the exception thrown an `ErrorException` or something else?

**Comment by joostvanhoof** [comment]
Nope, just the message that you see in the screenshot.

**Comment by riasvdv** [comment]
Do you have any logs you could share? Should be in `storage/logs/laravel.log`

**Comment by joostvanhoof** [comment]
Nothing shows up in the logs...

**Comment by riasvdv** [comment]
Could you temporarily change the try-catch by this snippet?

```php
        $unserialized = @unserialize($segmentClass);
        if ($unserialized !== false) {
            $segmentClass = $unserialized;
        }
```

If that fixes it then I'll add it to the next release, it seems like you can't try-catch unserialize errors

**Comment by joostvanhoof** [comment]
Thanks for looking into this. I have this running on production and I'm quite packed, is there a quick way to add this in production as this file is in the vendor folder?

**Comment by joostvanhoof** [comment]
Just a gentle reminder that I still get about 2 to 4 errors every time I create and send a campaign. As indicated in the previous comment I'm not too proficient in testing this in production so if you're able test, or have some instructions on how I can do this that would be much appreciated.

**Comment by riasvdv** [comment]
I will take a look at it again later this week

**Comment by riasvdv** [comment]
Should be fixed in 2.23.14
